scripts/audit_main_upload_buckets.py

```diff
@@ -0,0 +1,620 @@
+"""
+What's happening here?
+1. Input a Metamist project
+2. Get all the samples and sequences for this project from Metamist
+3. Check the bucket for all sequence data, and compare to the data in the metamist sequences
+    - If there are discrepencies, check if the file name and size is the same in the bucket as it is in metamist
+      - If they are both the same, assume this file has just been moved to a different location in the bucket
+      - Note the link between the Metamist sequence ID and this new bucket location
+      - Note the link between the Metamist sequence ID and the Sample ID
+
+4. Check if a sample has a completed cram - if so, its sequence data can be removed
+5. Remove the sequence data for samples with completed crams, including those whose data has been moved
+
+TODO: find other files to delete, not just fastqs
+TODO: make sure sequences are grabbed with the appropriate fields, e.g. read_type, project
+TODO: add filtering for exome vs genome
+TODO: add test and dry-run functionality
+TODO: make compatible with analysis-runner
+"""
+
+from collections import defaultdict
+import logging
+import click
+from cloudpathlib import CloudPath
+from cpg_utils.cloud import get_path_components_from_gcp_path
+from google.cloud import storage
+from sample_metadata.apis import (
+    ProjectApi,
+    ParticipantApi,
+    AnalysisApi,
+    SampleApi,
+    SequenceApi,
+)
+from sample_metadata.model.analysis_query_model import AnalysisQueryModel
+from sample_metadata.model.analysis_type import AnalysisType
+from sample_metadata.model.analysis_status import AnalysisStatus
+
+
+FASTQ_EXTENSIONS = ('.fq.gz', '.fastq.gz', '.fq', '.fastq')
+BAM_EXTENSIONS = ('.bam',)
+CRAM_EXTENSIONS = ('.cram',)
+GVCF_EXTENSIONS = ('.g.vcf.gz',)
+VCF_EXTENSIONS = ('.vcf', '.vcf.gz')
+ALL_EXTENSIONS = (
```

Comment:
You can add tuples, so would recommend just adding all the *_EXTENSIONS together